pkg/internal/reporter/reporter_test.go

```diff
@@ -20,16 +20,150 @@
 package reporter_test
 
 import (
+	"fmt"
+	"strconv"
 	"testing"
+	"time"
 
 	assert "github.com/stretchr/testify/require"
 
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	kconfigmap "github.com/kiagnose/kiagnose/kiagnose/configmap"
+
 	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/reporter"
 	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/status"
 )
 
+const (
+	testNamespace     = "target-ns"
+	testConfigMapName = "dpdk-checkup-config"
+)
+
 func TestReportShouldSucceed(t *testing.T) {
-	testReporter := reporter.New()
+	fakeClient := fake.NewSimpleClientset(newConfigMap())
+	testReporter := reporter.New(fakeClient, testNamespace, testConfigMapName)
 
 	assert.NoError(t, testReporter.Report(status.Status{}))
 }
+
+func TestReportShouldFailWhenCannotUpdateConfigMap(t *testing.T) {
+	// ConfigMap does not exist
+	fakeClient := fake.NewSimpleClientset()
+
+	testReporter := reporter.New(fakeClient, testNamespace, testConfigMapName)
+
+	assert.ErrorContains(t, testReporter.Report(status.Status{}), "not found")
+}
```

Comment:
Please consider moving this test after the success scenarios.

Reply:
Done